core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala

```diff
@@ -305,19 +305,12 @@ private[deploy] class SparkSubmitArguments(args: Seq[String], env: Map[String, S
   }
 
   private def validateKillArguments(): Unit = {
-    if (!master.startsWith("spark://") && !master.startsWith("mesos://")) {
-      error("Killing submissions is only supported in standalone or Mesos mode!")
-    }
```

Comment:
I am removing this part and although we dont fail fast we will fail when the new api will do the validation.